mono/mini/method-to-ir.c

```diff
@@ -1971,22 +1971,78 @@ mini_emit_memcpy (MonoCompile *cfg, int destreg, int doffset, int srcreg, int so
 	}
 }
 
-static void
-emit_tls_set (MonoCompile *cfg, int sreg1, MonoTlsKey tls_key)
+static MonoInst*
+mono_create_fast_tls_getter (MonoCompile *cfg, MonoTlsKey key)
 {
-	MonoInst *ins, *c;
+#ifdef MONO_ARCH_HAVE_FAST_TLS
+	int tls_offset = mono_tls_get_tls_offset (key);
 
-	if (cfg->compile_aot) {
-		EMIT_NEW_TLS_OFFSETCONST (cfg, c, tls_key);
-		MONO_INST_NEW (cfg, ins, OP_TLS_SET_REG);
-		ins->sreg1 = sreg1;
-		ins->sreg2 = c->dreg;
-		MONO_ADD_INS (cfg->cbb, ins);
-	} else {
+	if (cfg->compile_aot)
+		return NULL;
+
+	if (tls_offset != -1 && MONO_ARCH_HAVE_FAST_TLS) {
```

Comment:
Since we're renaming MONO_ARCH_HAVE_TLS_GET to MONO_ARCH_HAVE_FAST_TLS, let's remove this design mistake and simply make it a function that all backends have..